src/main/java/org/interstellarocean/terraforming/reflection/ReflectionModelTransformerProvider.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright © 2015 The Authors
+ *
+ * https://www.gnu.org/licenses/lgpl-3.0-standalone.html
+ */
+package org.interstellarocean.terraforming.reflection;
+
+import java.util.Collection;
+
+/**
+ * Provides builder allowing creation and configuration of {@link ReflectionModelTransformer} instances in a library-independent way.
+ * The provider implementation should isolate the library allowing easy change of used library
+ * in {@link org.interstellarocean.terraforming.ModelTransformer} implementations.
+ * <p>
+ * Suitable to implement the strategy pattern and for use with factory.
+ * </p>
+ *
+ * <h3>Example</h3>
+ *
+ * <blockquote><pre>
+ * public class FooModelTransformerProvider implements ReflectionModelTransformerProvider {
+ *
+ * 	private final FooFactory fooFactory;
+ *
+ * 	&#64;Inject // @Autowired, etc.
+ * 	public FooModelTransformerProvider(FooFactory fooFactory) {
+ * 		this.fooFactory = fooFactory;
+ * 	}
+ *
+ * 	&#64;Override
+ * 	public boolean provides(Class&lt;?&gt; library) {
+ * 		return org.bar.Foo.class.equals(library);
+ * 	}
+ *
+ * 	&#64;Override
+ * 	public &lt;SOURCE, DESTINATION&gt; Builder&lt;SOURCE, DESTINATION&gt; builder() {
+ * 		return sourceType -&gt; destinationType -&gt; excludeFieldNames -&gt; () -&gt; {
+ *
+ * 			FooConfigurator&lt;SOURCE, DESTINATION&gt; fooConfigurator = fooFactory.classTransformerConfigurator(sourceType, destinationType);
+ * 			excludeFieldNames.forEach(fieldName -&gt; fooConfigurator.exclude(fieldName));
+ * 			fooConfigurator.register();
+ *
+ * 			TransformerFacade&lt;SOURCE, DESTINATION&gt; reflectionModelTransformer = fooFactory.getTransformerFacade(sourceType, destinationType);
+ * 			return source -&gt; destination -&gt; {
+ * 				reflectionModelTransformer.transform(source, destination);
+ * 				return destination;
+ * 			};
+ *
+ * 		};
+ * 	}
+ *
+ * }
+ * </pre></blockquote>
+ *
+ * @see org.interstellarocean.terraforming.reflection
+ * @see ReflectionModelTransformer
+ * @see Builder
+ *
+ * @author Dariusz Wakuliński
+ */
+public interface ReflectionModelTransformerProvider {
+
+	/**
+	 * Allows selecting requested strategy by {@link ReflectionModelTransformerFactory}.
+	 *
+	 * @param library Class marking implementation of a strategy that use specific reflection transforming library.
+	 * 		Suitable for use by factory to select requested strategy.
+	 * 		Actually may be any class, but recommended is a class from the implemented library.
+	 * 		Should be unique across implementations (strategies).
+	 * @return True if implementation is suitable for requested library class.
+	 */
+	boolean provides(Class<?> library);
```

Comment:
Make it true strategy pattern. MT should not 'see' the providers. Factory should expose a `RMTFactory.builderFor(Class<?> library)` method and should select providers internally.
-1